saleor/core/tasks.py

```diff
@@ -34,7 +34,7 @@ def delete_event_payloads_task(expiration_date=None):
     payloads_to_delete = EventPayload.objects.filter(
         ~Exists(valid_deliveries.filter(payload_id=OuterRef("id")))
     ).order_by("-pk")
-    ids = payloads_to_delete.values_list("pk", flat=True)[:BATCH_SIZE]
+    ids = list(payloads_to_delete.values_list("pk", flat=True)[:BATCH_SIZE])
```

Comment:
Is that change required as well?

Reply:
I forgot to mention, this is a related, but separate bug.
Without it, the QuerySet is reevaluated on every access and can lead to a different set of payloads being fetched for file names and a different set for deletion